ucsb-cs156-f24/proj-happycows-f24-12
# Mr-OurTable testing - Reduced runtime and maintained testing accuracy

## Overview

In this PR, I rewrote the OurTable.test.js code to more efficiently test the ourTable.js and produce 100% mutation,line,and branch coverage. I sped up the testing by about 9 minutes.

## Screenshots (Optional)

Below is the original mutation testing runtime:
<img width="649" alt="OriginalMutationTime" src="https://github.com/user-attachments/assets/be856764-5e0e-4ee6-a3b2-cf69658d1d20">
Below is the new mutation testing runtime, still with 100%.
[
<img width="743" alt="NewMutationTime" src="https://github.com/user-attachments/assets/48a9c769-6b77-4f61-b148-50454b1eaa59">
](url)
Below is the new code's 100% branch/line coverage.
<img width="263" alt="Coverage" src="https://github.com/user-attachments/assets/5d0c6479-fb80-4270-9c8f-07a3eb67afc4">


## Feedback Request (Optional)

Please give suggestions on the following:
- I stuck with 100 rows for most of the testing. I'm not sure this is necessary, what do you think?

## Future Possibilities (Optional)

Some possible points could be:
- This is important in possibly testing new features for OurTable. The previous test suite ran really slowly so hopefully this will make testing more efficient.

## Validation (Optional)

1. Download this branch.
2. Run mutation testing for OurTable.js to see runtime/accuracy.
3. Run coverage testing for OurTable.js to see branch/line coverage.

## Tests


## Linked Issues

Closes #13
